pkg/apis/m3dboperator/v1/cluster.go

```diff
@@ -45,12 +45,12 @@ type M3DBClusterList struct {
 	Items           []M3DBCluster `json:"items"`
 }
 
-type CRDStatus struct {
-	State   CRDState `json:"state,omitempty"`
-	Message string   `json:"message,omitempty"`
+type M3DBStatus struct {
+	State   M3DBState `json:"state,omitempty"`
+	Message string    `json:"message,omitempty"`
 }
 
-type CRDState string
+type M3DBState string
```

Comment:
might be worth making enums values if are there a defined set of values this type can take - something like:

```
type M3DBState string

const (
  GreenState M3DBState ="green"
  YellowState M3DBState ="yellow"
  RedState M3DBState ="red"
  ...
)
```

Reply:
Yep, this was a rough commit that was recent. The idea is that there will entire status object that will define not only cluster status but node status as well. That object will be used to show the current state of the cluster holistically. If operator panics it will be able to read the status object to obtain the current state if any cluster/resource that kubernetes is aware of .